qubesadmin/tools/qvm_start_daemon.py

```diff
@@ -750,7 +750,14 @@ def main(args=None):
         kde=args.kde)
 
     if args.watch:
-        with daemon.pidfile.TimeoutPIDLockFile(args.pidfile):
+        fd = os.open(args.pidfile,
+                     os.O_WRONLY | os.O_CREAT | os.O_CLOEXEC,
+                     0o600)
+        with os.fdopen(fd, 'w') as lock_f:
+            fcntl.flock(fd, fcntl.LOCK_EX)
```

Comment:
`fcntl.LOCK_EX | fcntl.LOCK_NB`, and catch the possible exception to convert into appropriate error message?